examples/CMakeLists.txt

```
@@ -6,6 +6,7 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 add_subdirectory(hello_world)
+add_subdirectory(codegen)
```

Can we perhaps have this only enabled through a cmake option. I think we don't need to have to build each time.
```
option(UR_ENABLE_CODEGEN "Build the codegen example." OFF)
```

```suggestion
if(UR_ENABLE_CODEGEN)
    add_subdirectory(codegen)
endif()
```

---

Sure. Do you want an option per example or a generic `UR_ENABLE_EXAMPLES`?

---

My suggestion right would be to add `UR_BUILD_EXAMPLE_CODEGEN` just for this example to be consistent with other cmake options like this (`UR_BUILD_TOOLS`, `UR_BUILD_ADAPTER_...` and so on).